canonical-web-and-design/ubuntu.com
# Add /advantage/user-subscriptions endpoint

## Done

- Adding `tests.advantage.helpers` make function: `make_subscription`, `make_contract`. 
These will create mock instances of primitive classes. These will be used for testing in reproducing scenarios.
- Add `advantage.ua_contract.builders` functions: `build_user_subscriptions`, `build_initial_user_subscriptions`, `build_final_user_subscriptions`. These functions have the goal of transforming step by step primitives into models. In this case `UserSubscription` models.
- Add `advantage.ua_contract.helpers`  functions. These are smaller functions used by `builders` in the process of creating model objects. 
- Add `GET /advantage/user-subscriptions` endpoint

## QA

- Have an account with susbcriptions
- Login on (`?test_backend=true` is optional): https://ubuntu-com-10225.demos.haus
- Go to: https://ubuntu-com-10225.demos.haus/advantage/user-subscriptions 
- You should see json response of your user_subscriptions
- Use the view as feature and look at other user's "user subscriptions": https://ubuntu-com-10225.demos.haus/advantage/user-subscriptions?email=<email_address>


## Issue / Card

Fixes #https://github.com/canonical-web-and-design/commercial-squad/issues/184
